pkg/cliplugins/bind/plugin/bind.go

```diff
@@ -0,0 +1,174 @@
+	BindWaitTimeout time.Duration
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+
+	cmd.Flags().StringVar(&b.APIBindingName, "name", b.APIBindingName, "Name of the APIBinding to create.")
+	cmd.Flags().DurationVar(&b.BindWaitTimeout, "timeout", time.Second*30, "Duration to wait for APIBinding to be created successfully.")
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.APIExportRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.APIExportRef == "" {
+		return errors.New("`root:ws:apiexport_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.APIExportRef, "root") || !logicalcluster.New(b.APIExportRef).IsValid() {
+		return fmt.Errorf("fully qualified reference to workspace where APIExport exists is required. The format is `root:<ws>:<apiexport>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	workspacePath, apiExportName := logicalcluster.New(b.APIExportRef).Split()
+
+	// if apibindingName is not provided, default it to <apiExportname>.
+	apiBindingName := b.APIBindingName
+	if apiBindingName == "" {
+		apiBindingName = apiExportName
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
+	}
+
+	binding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: apiBindingName,
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       workspacePath.String(),
+					ExportName: apiExportName,
+				},
+			},
+		},
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	createdBinding, err := kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "apibinding %s created. Waiting to successfully bind ...\n", binding.Name); err != nil {
+		return err
+	}
+
+	// wait for phase to be bound
+	if createdBinding.Status.Phase != apisv1alpha1.APIBindingPhaseBound {
+		if err := wait.PollImmediate(time.Millisecond*500, b.BindWaitTimeout, func() (done bool, err error) {
```

Comment:
Why not set up a watch? Or, if we are going to poll, use one with a backoff so as to not spam requests.

Reply:
This was similar to what was done while [creating](https://github.com/kcp-dev/kcp/blob/12bbb4caf31e4c4bc67163e1d74418ffab7dcd76/pkg/cliplugins/workspace/plugin/kubeconfig.go#L568) worskspaces. We have configurable timeout and an interval of 0.5sec. Do you mean to create [`WatchEvent`](https://github.com/kubernetes/apimachinery/blob/70a38aaa19ef149d596adf1583696532c220f585/pkg/apis/meta/v1/watch.go#L31) for the APIBinding?

Reply:
You can use a client to watch the APIBinding, or use something from the `wait` package with backoff. Fine to defer for later to clean up all cli bits.